File: src/main/java/org/shark/example/service/mysql/FindTaskHistoryService.java
package org.shark.example.service.mysql;

import lombok.RequiredArgsConstructor;
import org.shark.example.dao.repository.TaskHistoryRepository;
import org.shark.example.dao.repository.pojo.TaskHistoryDo;
import org.springframework.stereotype.Service;

import java.util.List;

@RequiredArgsConstructor
@Service
public class FindTaskHistoryService {

    private final TaskHistoryRepository taskHistoryRepository;
    public List<TaskHistoryDo> start(Integer page, Integer size) {
        int offset = page * size;
        return taskHistoryRepository.findByPage(size, offset);
    }
}

File: src/main/kotlin/org/shark/example/service/mysql/FindTaskHistoryService.kt
package org.shark.example.service.mysql

import org.shark.example.dao.repository.TaskHistoryRepository
import org.shark.example.dao.repository.pojo.TaskHistoryDo
import org.springframework.stereotype.Service

@Service
class FindTaskHistoryService(
    private val taskHistoryRepository: TaskHistoryRepository
) {
    fun start(page: Int, size: Int): List<TaskHistoryDo> {
        val offset = page * size
        return taskHistoryRepository.findByPage(size, offset)
    }
}
